tardis/simulation/base.py

```diff
@@ -401,6 +406,137 @@ def to_hdf(self, model, path_or_buf, path='', plasma_properties=None):
+
+        #Just aliasing for cleaner expressions later
+        line_nu  = model.plasma.lines.nu
+        taus     = model.plasma.tau_sobolevs
+        att_S_ul = self.runner.att_S_ul
+        temps    = model.plasma.t_rad
+
+        dtau = 1 # Just to remember it 
+
+        L_nu  = np.zeros(nus.shape)
+
+        import pdb
+        for nu_idx,nu in enumerate(nus.value):
+            I_outer = np.zeros(ps_outer.shape)
+            for p_idx,p in enumerate(ps_outer):
+                z_cross_p = z_ct_outer[z_ct_outer[:,p_idx] > 0,p_idx]
+                z_cross_p = np.hstack((-z_cross_p,z_cross_p[::-1][1:],0)) # Zero ensures empty ks in last step below
+                                                                          # 1: avoids double counting center
+                shell_idx = (num_shell-1) - np.arange(n_shell_p_outer[p_idx]) # -1 for 0-based indexing
+                shell_idx = np.hstack((shell_idx,shell_idx[::-1][1:]))
+                
+                for idx,z_cross in enumerate(z_cross_p[:-1]):
+                    nu_start = nu / (1 + z_cross) 
+                    nu_end   = nu / (1 + z_cross_p[idx+1])
+                    shell = shell_idx[idx]
+                    # Note the direction of the comparisons
+                    ks, = np.where( (line_nu < nu_start) & (line_nu >= nu_end) )                    
+
+                    if len(ks) < 2:
+                        continue
+
+                    I_outer[p_idx] = dtau * ( 
+                                        ( J_rlues.iloc[ks[0],shell] + J_blues.iloc[ks[1],shell] ) / 2 )
+
```

Comment:
In any case you should comment them out if they aren't doing what you want them to do. I see several problems here.
`ks` could be anything and you are explicitly accessing `ks[0]` and `ks[1]` without knowing if there's more.
the way I understand it, `ks` is a mask of all the lines that have a matching nu. so accessing only one element looks suspicious to me